tools/metadata/index.ts

```diff
@@ -52,12 +52,10 @@ export async function scrapeToken(
   let provider
   if (typeof ethersProvider === 'string') {
     provider = new ethers.providers.JsonRpcProvider(ethersProvider)
-  } else if (typeof ethersProvider !== undefined) {
+  } else if (ethersProvider !== null) {
```

Comment:
```suggestion
  } else if (!ethersProvider) {
```

Because it could also be undefined.